server/src/main/java/org/elasticsearch/index/mapper/IgnoredFieldMapper.java

```diff
@@ -48,27 +67,71 @@ public String typeName() {
 
         @Override
         public Query existsQuery(SearchExecutionContext context) {
-            // This query is not performance sensitive, it only helps assess
-            // quality of the data, so we may use a slow query. It shouldn't
-            // be too slow in practice since the number of unique terms in this
-            // field is bounded by the number of fields in the mappings.
             return new TermRangeQuery(name(), null, null, true, true);
         }
 
         @Override
         public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
             return new StoredValueFetcher(context.lookup(), NAME);
         }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(FieldDataContext fieldDataContext) {
+            throw new IllegalArgumentException(
+                "aggregations on the '"
+                    + typeName()
+                    + "' field are supported for indices created by stack version "
+                    + AGGS_SUPPORT_VERSION
+                    + " or higher"
+            );
+        }
+    }
+
+    public static final class IgnoredFieldType extends StringFieldType {
+
+        private IgnoredFieldType() {
+            super(NAME, true, false, true, TextSearchInfo.SIMPLE_MATCH_ONLY, Collections.emptyMap());
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public Query existsQuery(SearchExecutionContext context) {
+            return new FieldExistsQuery(name());
```

Comment:
do we need to override this method? Can't we rely on the default implementation from MappedFieldType?

Reply:
This is in the legacy type so I don't think I added it...

Reply:
I think you should be able to remove it.